rest_framework/settings.py

```diff
@@ -187,6 +190,25 @@ def __getattr__(self, attr):
         except KeyError:
             # Fall back to defaults
             val = self.defaults[attr]
+        else:
+            # Verify that the user hasn't accidentally given a string instead
+            #  of an iterable, or vice-versa
+            default = self.defaults[attr]
+            if issubclass(val.__class__, Iterable) \
+                    and (not issubclass(default.__class__, Iterable)
+                         or isinstance(val, six.string_types)):
+                warnings.warn(
+                    "The `{attr}` setting must be iterable".format(**locals()),
```

Comment:
Don't use the '**locals()' style. Also more verbose might be good...

    'The "{settings_key}" setting must be a list or tuple, but got {type_name} with value "{value}".'